src/plugins/macaddr/macaddr.c

```diff
@@ -0,0 +1,271 @@
+		if (isdigit (*test++) == 0) return 1;
+	}
+	long macLong = atol (mac);
+
+	if (macLong > MAXMACINT || macLong < 0) return 1;
+
+	return 0;
+}
+
+int validateMac (Key * key)
+{
+	const Key * metaKey = keyGetMeta (key, "check/macaddr");
+	if (!metaKey) return 1;
+
+	const char * mac = keyString (key);
+
+	const char * regexColon = "^([0-9A-Fa-f]{2}:){5}([0-9A-Fa-f]{2})$";
+	const char * regexHyphen = "^([0-9A-Fa-f]{2}-){5}([0-9A-Fa-f]{2})$";
+	const char * regexHyphenSingle = "^([0-9A-Fa-f]{6}-)([0-9A-Fa-f]{6})$";
+
+	const char * regexStrings[] = { regexColon, regexHyphen, regexHyphenSingle };
+
+	int ret;
+	int i = 0;
+
+	ret = checkIntMac (mac);
+
+	while (ret == 1 && i < 3)
+	{
+		ret = checkRegex (mac, regexStrings[i]);
+		++i;
+	}
+
+	return ret;
+}
+
+void removeStandardizedSeparator (const char * mac, char * returned)
+{
+	int j = 0;
+	for (size_t i = 0; i < strlen (mac); i += 3)
+	{
+		returned[j++] = mac[i];
+		returned[j++] = mac[i + 1];
+	}
+}
+
+void transformToInt (Key * key)
+{
+	const char * mac = keyString (key);
+
+	char * macReduced = elektraMalloc (12);
+
+	removeStandardizedSeparator (mac, macReduced);
+
+	long intValue = strtol (macReduced, NULL, 16);
+
+	const int n = snprintf (NULL, 0, "%lu", intValue);
+	char * buffer = elektraMalloc (n + 1);
+	snprintf (buffer, n + 1, "%lu", intValue);
+
+	keySetString (key, buffer);
+	elektraFree (macReduced);
+	elektraFree (buffer);
+}
+
+int elektraMacaddrGet (Plugin * handle ELEKTRA_UNUSED, KeySet * returned, Key * parentKey)
+{
+	if (!elektraStrCmp (keyName (parentKey), "system/elektra/modules/macaddr"))
+	{
+		KeySet * contract =
+			ksNew (30, keyNew ("system/elektra/modules/macaddr", KEY_VALUE, "macaddr plugin waits for your orders", KEY_END),
+			       keyNew ("system/elektra/modules/macaddr/exports", KEY_END),
+			       keyNew ("system/elektra/modules/macaddr/exports/get", KEY_FUNC, elektraMacaddrGet, KEY_END),
+			       keyNew ("system/elektra/modules/macaddr/exports/set", KEY_FUNC, elektraMacaddrSet, KEY_END),
+
+#include ELEKTRA_README
+
+			       keyNew ("system/elektra/modules/macaddr/infos/version", KEY_VALUE, PLUGINVERSION, KEY_END), KS_END);
+		ksAppend (returned, contract);
+		ksDel (contract);
+
+		return ELEKTRA_PLUGIN_STATUS_SUCCESS;
+	}
+	ksRewind (returned);
+	Key * cur;
+	while ((cur = ksNext (returned)) != NULL)
+	{
+		const Key * meta = keyGetMeta (cur, "check/macaddr");
+		if (!meta) continue;
+		const Key * origValue = keyGetMeta (cur, "origvalue");
+		if (origValue)
+		{
+			ELEKTRA_SET_ERRORF (
+				ELEKTRA_ERROR_STATE, parentKey,
+				"Meta key 'origvalue' for key %s not expected to be set, another plugin has already set this meta key!",
+				keyString (cur));
+			return ELEKTRA_PLUGIN_STATUS_ERROR;
+		}
+		int rc = validateMac (cur);
+		if (rc)
+		{
+			ELEKTRA_SET_ERRORF (ELEKTRA_ERROR_INVALID_FORMAT, parentKey, "%s is not in a supported format.", keyString (cur));
+			return ELEKTRA_PLUGIN_STATUS_ERROR;
+		}
+
+		if (checkIntMac (keyString (cur)))
```

Comment:
You already call `checkIntMac` inside `validateMac`, if you use a different return code, for the case where an integer mac was given, you won't have to call it again here.